src/iter.rs

```diff
@@ -29,7 +29,8 @@ impl<'a> HexToBytesIter<HexDigitsIter<'a>> {
     ///
     /// If the input string is of odd length.
     #[inline]
-    pub fn new(s: &'a str) -> Result<Self, OddLengthStringError> {
+    #[allow(dead_code)]
```

Comment:
In 548aa04953a8c63bb99a83c58a765b7b03966e3e:

We should just delete the function if it's not used.

Reply:
But we will want it back soon as we make this type public, so I figured leave it in there. Right?

Reply:
Added a comment to the attribute.

Reply:
Eh, I guess this is fine. We'll remember because we talked about it here.